### PR TITLE
Update Action workflows

### PR DESCRIPTION
**Pull request by triffer**
**Description**

Changes proposed in this pull request:

- Update actions workflows since our PRs use the workflows from main and only by having the update workflows in main we can make sure that the mod-dev merge branch can execute all workflows.

**Related issue(s)**

### CODE REVIEW

File: .github/actions/upgrade-test/action.yaml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    - name: Deploy prerequisite components
+      shell: bash
+      run: |
+        KUBECONFIG=$(k3d kubeconfig merge kyma) kyma deploy -s main -c hack/kyma-components.yaml
+    - name: Deploy mod-dev version of manager #this step should be removed after first api gateway manager release. This behavior should be moved to for example some deploy-latest-release-script
+      shell: bash
+      env:
+        IMG: ${{ inputs.manager_image}}
+      run: |
+        git checkout mod-dev
```
Comment (barchw) [inline .github/actions/upgrade-test/action.yaml:47]:
```suggestion
        git checkout main
```

**Review by kolodziejczak — DISMISSED** [review]

File: .github/workflows/main-integration.yaml [hunk 2]
```diff
@@ -1,22 +1,18 @@
...
       - "docs/**"
       - "**/*.md"
-      - "tests/performance/**"
-      - "OWNERS"
       - "CODEOWNERS"
       - "sec-scanners-config.yaml"
       - ".github/**" #Ignoring since build job isn't triggered on the workflow directory changes
     branches:
       - main
-  schedule:
```
Comment (barchw) [inline .github/workflows/main-integration.yaml:18]: Maybe we should comment it out just like the upgrade job?

**Review by kolodziejczak — APPROVED** [review]

**Review by barchw — APPROVED** [review]

**Pull request merged**